File: app/src/main/java/com/varunbarad/popularmovies/model/data/CompanyStub.java
package com.varunbarad.popularmovies.model.data;

import com.google.gson.Gson;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Creator: vbarad
 * Date: 2016-12-07
 * Project: PopularMovies
 */

public final class CompanyStub {
  @Expose
  @SerializedName("id")
  private long id;
  @Expose
  @SerializedName("name")
  private String name;

  /**
   * No args constructor for serialization
   */
  public CompanyStub() {

  }

  /**
   * @param id
   * @param name
   */
  public CompanyStub(final long id, final String name) {
    this.id = id;
    this.name = name;
  }

  /**
   * @return The unique id for this Genre
   */
  public final long getId() {
    return id;
  }

  /**
   * @return The name for this Genre
   */
  public final String getName() {
    return name;
  }

  @Override
  public String toString() {
    return (new Gson()).toJson(this);
  }
}

File: app/src/main/java/com/varunbarad/popularmovies/model/data/CompanyStub.kt
package com.varunbarad.popularmovies.model.data

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import com.squareup.moshi.Moshi

/**
 * Creator: Varun Barad
 * Date: 2019-06-05
 * Project: PopularMovies
 */
@JsonClass(generateAdapter = true)
data class CompanyStub(
    @Json(name = "id") val id: Long,
    @Json(name = "name") val name: String = ""
) {
    override fun toString(): String {
        return Moshi.Builder().build().adapter(CompanyStub::class.java).toJson(this)
    }
}
